### PR TITLE
[report-converter] Support Markdownlint

### PR DESCRIPTION
**Pull request by csordasmarton**

### CODE REVIEW

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- tools/report-converter/codechecker_report_converter/markdownlint/analyzer_result.py  2
- tools/report-converter/codechecker_report_converter/markdownlint/output_parser.py  3
- tools/report-converter/tests/unit/test_mdl_parser.py  3
         

Clones added
============
- tools/report-converter/codechecker_report_converter/markdownlint/output_parser.py  2
- tools/report-converter/tests/unit/test_mdl_parser.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5833787&bid=19101450)

File: tools/report-converter/codechecker_report_converter/markdownlint/output_parser.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        self.analyzer_result = analyzer_result
+
+        # Regex for parsing a clang-tidy message.
+        self.message_line_re = re.compile(
+            # File path followed by a ':'.
+            r'^(?P<path>[\S ]+?):'
+            # Line number followed by a ':'.
+            r'(?P<line>\d+?): '
+            # Checker name.
+            r'(?P<checker>[\S ]+) '
```
Comment (bruntib) [inline tools/report-converter/codechecker_report_converter/markdownlint/output_parser.py:33]: Does space character need to be included in checker name? Doesn't it capture all words of the checker message eagerly except for the last one?

**Review by bruntib — CHANGES_REQUESTED** [review]

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- tools/report-converter/codechecker_report_converter/markdownlint/output_parser.py  3
- tools/report-converter/codechecker_report_converter/markdownlint/analyzer_result.py  2
- tools/report-converter/tests/unit/test_mdlint_parser.py  3
         

Clones added
============
- tools/report-converter/codechecker_report_converter/markdownlint/output_parser.py  2
- tools/report-converter/tests/unit/test_mdlint_parser.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5833787&bid=19101450)

File: tools/report-converter/README.md [hunk 2]
```diff
@@ -362,6 +364,28 @@ report-converter -t golint -o ./codechecker_golint_reports ./golint_reports.out
...
+The following example shows you how to run Markdownlint and store the results
+found by Markdownlint to the CodeChecker database.
+
+```sh
+# Run Markdownlint.
+mdl /path/to/your/project > ./mdlint_reports.out
+
+# Use 'report-converter' to create a CodeChecker report directory from the
+# analyzer result of Markdownlint.
+report-converter -t mdlint -o ./codechecker_mdlint_reports ./mdlint_reports.out
```
Comment (gyorb) [inline tools/report-converter/README.md:383]:
```suggestion
report-converter -t mdl -o ./codechecker_mdl_reports ./mdl_reports.out
```

File: tools/report-converter/README.md [hunk 3]
```diff
@@ -362,6 +364,28 @@ report-converter -t golint -o ./codechecker_golint_reports ./golint_reports.out
...
+
+The recommended way of running Markdownlint is to redirect the output to a file
+and give this file to the report converter tool.
+
+The following example shows you how to run Markdownlint and store the results
+found by Markdownlint to the CodeChecker database.
+
+```sh
+# Run Markdownlint.
+mdl /path/to/your/project > ./mdlint_reports.out
```
Comment (gyorb) [inline tools/report-converter/README.md:379]:
```suggestion
mdl /path/to/your/project > ./mdl_reports.out
```

File: tools/report-converter/README.md [hunk 4]
```diff
@@ -362,6 +364,28 @@ report-converter -t golint -o ./codechecker_golint_reports ./golint_reports.out
...
+
+# Use 'report-converter' to create a CodeChecker report directory from the
+# analyzer result of Markdownlint.
+report-converter -t mdlint -o ./codechecker_mdlint_reports ./mdlint_reports.out
+
+# Store Markdownlint reports with CodeChecker.
+CodeChecker store ./codechecker_mdlint_reports -n mdlint
```
Comment (gyorb) [inline tools/report-converter/README.md:386]:
```suggestion
CodeChecker store ./codechecker_mdl_reports -n mdl
```

File: tools/report-converter/codechecker_report_converter/markdownlint/analyzer_result.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+from codechecker_report_converter.analyzer_result import AnalyzerResult
+
+from .output_parser import MarkdownlintParser
+from ..plist_converter import PlistConverter
+
+
+class MarkdownlintAnalyzerResult(AnalyzerResult):
+    """ Transform analyzer result of Markdownlint. """
+
+    TOOL_NAME = 'mdlint'
```
Comment (gyorb) [inline tools/report-converter/codechecker_report_converter/markdownlint/analyzer_result.py:18]:
```suggestion
    TOOL_NAME = 'mdl'
```

File: tools/report-converter/tests/unit/mdlint_output_test_files/Makefile [hunk 6]
```diff
@@ -0,0 +1,2 @@
+all:
+	mdl ./files/readme.md > readme.out
```
Comment (gyorb) [inline tools/report-converter/tests/unit/mdlint_output_test_files/Makefile:2]: maybe it is worth to mention here or somewhere in the test documentation that there is a docker file which can be used to install and execute markdownlint easier:
```
docker run --rm -v ${PWD}:/data markdownlint/markdownlint .
```

**Review by gyorb — CHANGES_REQUESTED** [review]
There is an extra dot in the store file paths:
~/codechecker/docs**/.**/usage.md

I think it gets there during the report conversion because the parse command prints out that extra dot too in the filepath.

**Comment by gyorb** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- tools/report-converter/tests/unit/test_mdl_parser.py  3
- tools/report-converter/codechecker_report_converter/markdownlint/analyzer_result.py  2
- tools/report-converter/codechecker_report_converter/markdownlint/output_parser.py  3
         

Clones added
============
- tools/report-converter/tests/unit/test_mdl_parser.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/CodeChecker/codechecker/pullRequest?prid=5833787&bid=19101450)

**Review by bruntib — APPROVED** [review]

**Pull request merged**